src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,24 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.CantAddUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws CantAddUserException {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new CantAddUserException("User already exists in system");
+        } else if (user.getLogin().length() < 6) {
```

Comment:
Create constants for 6 and 18
Check if login, password and age == null
Check if login and password == ""
Check if user == null